src/Time/TimeSteppers/DormandPrince5.hpp

```diff
@@ -0,0 +1,260 @@
+  const double dt = time_step.value();
+
+  switch (substep) {
+    case 0: {
+      *u += (a2 * dt) * history->begin().derivative();
+      break;
+    }
+    case 1: {
+      *u = u0;
+      for (size_t i = 0; i < 2; ++i) {
+        *u += (gsl::at(a3, i) * dt) *
+              (history->begin() + static_cast<int>(i)).derivative();
+      }
+      break;
+    }
+    case 2: {
+      *u = u0;
+      for (size_t i = 0; i < 3; ++i) {
+        *u += (gsl::at(a4, i) * dt) *
+              (history->begin() + static_cast<int>(i)).derivative();
+      }
+      break;
+    }
+    case 3: {
+      *u = u0;
+      for (size_t i = 0; i < 4; ++i) {
+        *u += (gsl::at(a5, i) * dt) *
+              (history->begin() + static_cast<int>(i)).derivative();
+      }
+      break;
+    }
+    case 4: {
+      *u = u0;
+      for (size_t i = 0; i < 5; ++i) {
+        *u += (gsl::at(a6, i) * dt) *
+              (history->begin() + static_cast<int>(i)).derivative();
+      }
+      break;
+    }
+    case 5: {
+      *u = u0;
+      for (size_t i = 0; i < 6; ++i) {
+        *u += (gsl::at(b, i) * dt) *
+              (history->begin() + static_cast<int>(i)).derivative();
+      }
+      break;
+    }
+    default:
+      ERROR("Substep in DP5 should be one of 0,1,2,3,4,5, not " << substep);
+  }
+
+  // Clean up old history
+  if (history->size() == number_of_substeps()) {
+    history->mark_unneeded(history->end());
+  }
+}
+
+template <typename Vars, typename DerivVars>
+void DormandPrince5::dense_update_u(const gsl::not_null<Vars*> u,
+                                    const History<Vars, DerivVars>& history,
+                                    const double time) const noexcept {
+  ASSERT(history.size() == number_of_substeps(),
+         "DP5 can only dense output on last substep ("
+             << number_of_substeps() - 1 << "), not substep "
+             << history.size() - 1);
+  const double t0 = history[0].value();
+  const double t_end = history[history.size() - 1].value();
+  // The history does not contain the final step; specifically,
+  // step_end = t0 + c[4] * dt, so dt = (t_end - t0) / c[4]. But since
+  // c[4] = 1.0 for DP5, we don't need to divide by c[4] here.
+  const double dt = t_end - t0;
+  const double output_fraction = (time - t0) / dt;
+  ASSERT(output_fraction >= 0.0, "Attempting dense output at time "
+                                     << time << ", but already progressed past "
+                                     << t0);
+  ASSERT(output_fraction <= 1.0, "Requested time ("
+                                     << time << ") not within step [" << t0
+                                     << ", " << t0 + dt << "]");
+
+  // Get the solution u1 at the final time
+  const auto& u0 = history.begin().value();
+  Vars u1 = u0;
+  for (size_t i = 0; i < 6; ++i) {
+    u1 += (gsl::at(b, i) * dt) *
+          (history.begin() + static_cast<int>(i)).derivative();
+  }
+
+  // I will need the following: k1, k3, k4, k5, k6.
```

Comment:
"I will" -> "We" so that whenever someone reads it in the future it feels less tied to a specific person.